core/src/main/scala/org/apache/spark/resource/ResourceUtils.scala

```diff
@@ -70,6 +86,258 @@ private[spark] object ResourceUtils extends Logging {
   // internally we currently only support addresses, so its just an integer count
   val AMOUNT = "amount"
 
+  /**
+   * Assign resources to workers from the same host to avoid address conflict.
```

Comment:
workers/drivers.  It would be nice to add a bit more description here about file locking and what all this does.